src/main/java/io/stargate/sgv2/jsonapi/service/embedding/configuration/EmbeddingProviderResponseValidation.java

```diff
@@ -36,19 +45,35 @@ public void filter(ClientRequestContext requestContext, ClientResponseContext re
     if (responseContext.getStatus() == 0) {
       return;
     }
+    // log the whole error body here
+    String responseBody = null;
```

Comment:
Is this called only for error responses? From name it'd sound it was called for every response, error and non-error: if so, we should only collect and log response in fail case (4xx or 5xx response).

Reply:
Yes, I was moving too fast and forgot this